idea/idea-fir/src/org/jetbrains/kotlin/idea/quickfix/fixes/WrapWithSafeLetCall.kt

```diff
@@ -0,0 +1,290 @@
+        val calleeName = callExpression.calleeExpression?.text ?: return null
+        val callSite = callExpression.parent as? KtQualifiedExpression ?: callExpression
+        val functionalVariableSymbol = (callExpression.resolveCall() as? KtVariableWithInvokeFunctionCall)?.target ?: return false
+        val localScope = callExpression.containingKtFile.getScopeContextForPosition(callSite)
+        // If no symbol in the local scope contains the called symbol, then the symbol must be a member symbol.
+        return localScope.scopes.getCallableSymbols { it.identifierOrNullIfSpecial == calleeName }.any { it == functionalVariableSymbol }
+    }
+
+    val forUnsafeInfixCall = diagnosticFixFactory<KtFirDiagnostic.UnsafeInfixCall> { diagnostic ->
+        createWrapWithSafeLetCallInputForNullableExpressionIfMoreThanImmediateParentIsWrapped(diagnostic.receiverExpression)
+    }
+
+    val forUnsafeOperatorCall = diagnosticFixFactory<KtFirDiagnostic.UnsafeOperatorCall> { diagnostic ->
+        createWrapWithSafeLetCallInputForNullableExpressionIfMoreThanImmediateParentIsWrapped(diagnostic.receiverExpression)
+    }
+
+    val forArgumentTypeMismatch = diagnosticFixFactory<KtFirDiagnostic.ArgumentTypeMismatch> { diagnostic ->
+        if (diagnostic.isMismatchDueToNullability) createWrapWithSafeLetCallInputForNullableExpression(diagnostic.psi.wrappingExpressionOrSelf)
+        else emptyList()
+    }
+
+    private fun KtAnalysisSession.createWrapWithSafeLetCallInputForNullableExpressionIfMoreThanImmediateParentIsWrapped(
+        nullableExpression: KtExpression?,
+        isImplicitInvokeCallToMemberProperty: Boolean = false,
+    ): List<IntentionAction> {
+        val surroundingExpression = nullableExpression?.typableParentExpression
+        if (
+            surroundingExpression == null ||
+            // If the surrounding expression is at a place that accepts null value, then we don't provide wrap with let call because the
+            // plain safe call operator (?.) is a better fix.
+            isExpressionAtNullablePosition(surroundingExpression) ||
+            // In addition, if there is no parent that is at a nullable position, then we don't offer wrapping with let either because
+            // it still doesn't fix the code. Hence, the plain safe call operator is a better fix.
+            findParentExpressionAtNullablePosition(nullableExpression) == null
+        ) {
+            return emptyList()
+        }
+        return createWrapWithSafeLetCallInputForNullableExpression(nullableExpression, isImplicitInvokeCallToMemberProperty)
+    }
+
+    private fun KtAnalysisSession.createWrapWithSafeLetCallInputForNullableExpression(
+        nullableExpression: KtExpression?,
+        isImplicitInvokeCallToMemberProperty: Boolean = false,
+        parentExpression: KtExpression? = findParentExpressionAtNullablePosition(nullableExpression)
```

Comment:
No argument is ever passed for this parameter so this should be a local variable inside the function

Reply:
...or I think you meant to pass the value of `findParentExpressionAtNullablePosition(nullableExpression)` from `createWrapWithSafeLetCallInputForNullableExpressionIfMoreThanImmediateParentIsWrapped()` so you don't have to compute it twice?

Reply:
It's actually an artifact during refactoring. But it's a good point to save an additional call so let me do that.